crates/wasmparser/src/validator/types.rs

```diff
@@ -2438,6 +2612,9 @@ impl TypeList {
             component_funcs: component_funcs.len(),
             core_modules: core_modules.len(),
             core_instances: core_instances.len(),
+            core_type_to_rec_group: core_type_to_rec_group.len(),
+            rec_group_elements: rec_group_elements.len(),
+            canonical_rec_groups: canonical_rec_groups.clone(),
```

Comment:
This is cloning an entire hash table on all checkpoints, right? That seems like it could get expensive?

Reply:
Oh I remember we talked a bit about asserting that the hash table just doesn't change during the checkpoint and I suppose that didn't make its way here yet